service/accounting/db/query.sql

```diff
@@ -34,10 +34,18 @@ RETURNING *;
 SELECT id, company_id, branch_id, account_code, account_name, account_group, bank_name, 
 bank_account_number, bank_code, opening_balance, is_deleted
 FROM accounting.company_chart_of_accounts
-WHERE company_id = $1 AND account_name LIKE $2
+WHERE company_id = $1 
+AND CASE 
+WHEN $5 = 'DEBET' THEN 
```

Comment:
ini query nya kok bloated banget ya? keep the query simple, logic nya taruh di aplikasi nya / golang nya. (all business logic expected nya taruh di golangnya)

Reply:
Kalau gini apa query nya dipisah ya jadinya?

Reply:
engga perlu dipisah

Reply:
`SELECT id, company_id, branch_id, account_code, account_name, account_group, bank_name, bank_account_number, bank_code, opening_balance, is_deleted
FROM accounting.company_chart_of_accounts
WHERE company_id = $1 
AND account_name = $2
AND CASE WHEN @is_filter_account_group::bool THEN account_group IN $3::TEXT[] ELSE TRUE END
AND CASE WHEN @is_deleted_filter::bool THEN is_deleted = $4 ELSE TRUE END
END;`

1. ini aku ada query tolong dicek sesuai dengan yang tujuan diatas atau engga, 

2. jadi tujuannya dengan query ini, sql nya engga perlu tahu oh ada KREDIT / DEBET lagi, 

kalau di request ada DEBET brati langsung lempar array dari golang nya ['KAS', 'BANK']
kalau di request ada KREDIT brati langsung lempar array dari golang nya ['BEBAN USAHA', 'BEBAN LAIN-LAIN']

Reply:
Ok udah paham pakai IN di bawah ya, aku belum explore terlalu banyak di sqlc nya soalnya

Reply:
Ternyata bisa pass array di sqlc